### ISSUE TITLE
Document expected "map" behavior of KeyValueList

### ISSUE THREAD

**Issue by tigrannajaryan** [open-telemetry/opentelemetry-proto#296]
KeyValueList can technically represent a multi-map. However, the Otel API semantics require unique keys. We need to document the behavior in case the receivers see non-unique keys.

**Comment by jsuereth** [comment]
Is it acceptable to call this an "error state" and allow implementations to choose how they handlje it (with a minimum of reporting the issue somehow), or do we want a consistent story here?  I can see (performance) reasons to let impelmentors pick how to deal with that scenario.

**Comment by tigrannajaryan** [comment]
> Is it acceptable to call this an "error state" and allow implementations to choose how they handlje it (with a minimum of reporting the issue somehow), or do we want a consistent story here? I can see (performance) reasons to let impelmentors pick how to deal with that scenario.

I agree. Precisely for performance reasons. Otherwise we would need to validate this in the Collector otlp reciever which would be quite costly.

**Comment by tigrannajaryan** [comment]
The uniqueness requirement was documented in https://github.com/open-telemetry/opentelemetry-proto/pull/350

I don't think we need to say anything more. We deliberately avoid over-specifying the receiver behavior to allow receivers to optimize for performance assuming the sender is following the requirements (so that the receiver does not have to do costly uniqueness checks).
